### PR TITLE
Added custom-templates

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary
This PR add support for rendering custom-templates in eos_cli_config_gen.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Component(s) name
custom-templates

## Proposed changes

Templates should be placed in a path below playbook and referenced as a list in the variable `custom_templates`.
This also works with eos_l3ls_evpn role since the variable custom_templates passes right through.

The purpose of custom-templates is for users to add extra functionality to avd without breaking existing functionality or modifying avd roles. The templates will render last, so they can override any other configuration.

Readme.md for eos_cli_config_gen has been updated.
Device documentation has been updated to show the rendered result for each custom template.

## How to test

Referenced one j2 template on DC-level and two for all Spines.
One of them with static content and the other with j2 dynamically rendered configuration.

## Checklist:

- [?] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [ ] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/custom-templates.j2 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% if custom_templates is defined and custom_templates|length > 0 %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/custom-templates.j2]: please refactor to: {% if custom_templates is defined and custom_templates is not none %}

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/custom-templates.j2 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{% if custom_templates is defined and custom_templates|length > 0 %}
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/custom-templates.j2]: please refactor to: {% if custom_templates is defined and custom_templates is not none %}

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
@ClausHolbechArista please see in-line comments, more about code arista.avd coding standards.

**Comment by ClausHolbechArista** [comment]
Because of https://github.com/ansible/ansible/issues/7106 we cannot access templates outside of roles or playbook directory. This means that custom templates must be placed in a directory below the roles or below the playbook.
Symlinks are allowed, so if playbook is placed in `<root dir>/playbooks/` and templates are placed in `<root dir>/custom/` you can add a symlink `/playbooks/custom -> ../custom/`

I will update readme to reflect this constraint.

**Review by carlbuchmann — APPROVED** [review]
Validated workflow with both CloudVision and eAPI deployment methods.

I also tested using a symlink, since my custom template folder was not directly under my playbooks folder.

**Comment by c-po** [comment]
The `custom_templates` directive is really interesting.

I see all templates are simply concatenated. Would it probably make sense to add a `!` delimiter between individual configlets by default when rendering it? That would ease debugging using my visual EOS interpreter located in my brain.

**Pull request merged**